### ISSUE TITLE
[WAGON-57] Error deploying 3rd party jar to company webdav repository

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-wagon#224]
**[svarog](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=svarog)** opened **[WAGON-57](https://issues.apache.org/jira/browse/WAGON-57?redirect=false)** and commented

Deploying activation jar to company webdav repository fails

mvn -X deploy:deploy-file -DgroupId=javax.activation -DartifactId=activation -Dversion=1.0.2 -Dpackaging=jar -Dfile=activation.jar -DrepositoryId=interseek-maven-repository -Durl=dav:https://maven.noviforum.si

Error deploying artifact: Unsupported Protocol: 'dav': Cannot find wagon which supports the requested protocol: dav

See atached file for complete console output of the error.

I suspect version conflicts. My coworker on linux succeded at deploying some other sun jar on the same company repo. The documentation says:
First, the wagon-provider(wagon-ftp, wagon-file, etc..) must be placed to your %M2_HOME%/lib.

I copied wagon-webdav-1.0-beta-1.jar there, but it doesn't help.

---

**Affects:** 1.0-beta-1

**Attachments:**
- [output.txt](https://issues.apache.org/jira/secure/attachment/12720762/output.txt) (_15.63 kB_)

**Comment by jira-importer** [comment]
**[svarog](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=svarog)** commented

I copied

jakarta-slide-webdavlib-2.1.jar
commons-httpclient-3.0.jar

Now I am getting

[ERROR] BUILD ERROR
[INFO] ------------------------------------------------------------------------
[INFO] Error deploying artifact: Unsupported Protocol: 'dav': Cannot find wagon which supports the requested protocol: dav

(class: org/apache/maven/wagon/providers/webdav/WebDavWagon, method: openConnection signature: ()V) Incompatible object argument f
or function call

**Comment by jira-importer** [comment]
**[Carlos Sanchez Gonzalez](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlos)** commented

If you look the pom http://www.ibiblio.org/maven2/org/apache/maven/wagon/wagon-webdav/1.0-beta-1/wagon-webdav-1.0-beta-1.pom

you'll see

\<dependency>
\<groupId>slide\</groupId>
\<artifactId>slide-webdavlib\</artifactId>
\<version>2.1\</version>
\</dependency>

which in turn depends on

\<dependency>
\<groupId>commons-httpclient\</groupId>
\<artifactId>commons-httpclient\</artifactId>
\<version>2.0.2\</version>
\</dependency>
not 3.0

**Comment by jira-importer** [comment]
**[svarog](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=svarog)** commented

Thanks for your input Carlos.

I added one more jar (commons-logging) and it works now.

During upload I got lots of this messages (INFO: Redirect requested but followRedirects is disabled)

Uploading: https://maven.noviforum.si/javax/activation/activation/1.0.2/activation-1.0.2.jar
26.7.2006 22:09:49 org.apache.commons.httpclient.HttpMethodBase processRedirectResponse
INFO: Redirect requested but followRedirects is disabled
26.7.2006 22:09:49 org.apache.commons.httpclient.HttpMethodBase processRedirectResponse
INFO: Redirect requested but followRedirects is disabled
26.7.2006 22:09:49 org.apache.commons.httpclient.HttpMethodBase processRedirectResponse
INFO: Redirect requested but followRedirects is disabled
26.7.2006 22:09:49 org.apache.commons.httpclient.HttpMethodBase processRedirectResponse
INFO: Redirect requested but followRedirects is disabled
26.7.2006 22:09:49 org.apache.commons.httpclient.HttpMethodBase processRedirectResponse
......

Is there some setting to enable redirects? Anyway, the upload works - activation jar and accompanied files are all there in internal repo.
